vissl/utils/wandb.py

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+This script contains some helpful functions to handle wandb setup.
+"""
+
+import os
+import logging
+
+from classy_vision.generic.distributed_util import is_primary
+from vissl.utils.checkpoint import get_checkpoint_folder
+from vissl.utils.io import makedir
+from omegaconf import DictConfig
+
+def is_wandb_available():
+    """
+    Check whether wandb is available or not.
+
+    Returns:
+        wandb_available (bool): based on wandb imports, returns whether tensboarboard
+                             is available or not.
+    """
+    try:
+        import wandb  # noqa F401
+
+        wandb_available = True
+    except ImportError:
+        logging.info("Tensorboard is not available")
+        wandb_available = False
+    return wandb_available
+
+
+def get_wandb_dir(cfg):
+    """
+    Get the output directory where the wandb events will be written.
+
+    Args:
+        cfg (AttrDict): User specified config file containing the settings for the
+                        wandb as well like log directory, logging frequency etc
+
+    Returns:
+        wandb_dir (str): output directory path
+
+    """
+    checkpoint_folder = get_checkpoint_folder(cfg)
+    wandb_dir = f"{checkpoint_folder}/wandb_logs"
+    logging.info(f"Wandb Local dir: {wandb_dir}")
+    makedir(wandb_dir)
+    return wandb_dir
+
+
+def parse_hydra_config(cfg):
+    final = {}
+
+    def recurse(key_array):
+        current = cfg
+        for key in key_array:
+            current = getattr(current, key)
+
+        if type(current) != DictConfig:
+            final['.'.join(key_array)] = current
+        else:
+            for key in current:
+                recurse(key_array + [key])
+
+    recurse([])
+    return final
+
+
+def get_wandb_hook(cfg):
+    """
+    Construct the Tensorboard hook for visualization from the specified config
+
+    Args:
+        cfg (AttrDict): User specified config file containing the settings for the
+                        wandb as well like log directory, logging frequency etc
+
+    Returns:
+        SSLTensorboardHook (function): the wandb hook constructed
+    """
+    import wandb
+    from vissl.hooks import SSLWandbHook
```

Comment:
move these to the top or is it due to circular imports?

Reply:
I want to only import wandb if the program is calling the hook. This way it will still work if a user has not installed wandb yet.

Reply:
makes sense to me to do local imports .